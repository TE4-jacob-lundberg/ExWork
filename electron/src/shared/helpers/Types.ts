export interface ILink {
  url: string;
  label: string;
}

export interface IGameFormFields {
  id: string;
  title: string;
  abbreviation: string;
  image: string; 
  xAxis: string;
  yAxis: string;
  links: ILink[];
  fileNames: string[];
}
