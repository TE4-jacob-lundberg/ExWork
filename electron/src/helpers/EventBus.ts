/* eslint-disable @typescript-eslint/no-explicit-any */
const EventBus: IEventBus = {
  events: {},
  dispatch: function (event: string, data: any): void {
    if (!this.events[event]) return;
    this.events[event].forEach((callBack: Function) => callBack(data));
  },
  subscribe: function (event: string, callBack: Function): void {
    if (!this.events[event]) this.events[event] = [];
    this.events[event].push(callBack);
  },
};

interface IEventBus {
  events: {[id: string]: Function[]};
  dispatch: (event: string, data: any) => void;
  subscribe: (event: string, callBack: Function) => void;
}

export { EventBus };
