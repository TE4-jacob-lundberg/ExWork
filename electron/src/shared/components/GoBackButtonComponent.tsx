import React from 'react';
import { useHistory } from 'react-router-dom';
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { css, jsx } from '@emotion/core';
import Styled from '@emotion/styled';

/* @jsx jsx */

interface Props {}

const GoBackButtonComponent: React.FC<Props> = function () {
  const history = useHistory();

  const ContainerStyled = Styled.div`
    position: absolute;
    z-index: 9999px;
    padding: 8px;
    border-radius: 5px;
    left: 5%;
    top: 5%;
    height: 50px;
    background: #333;
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: default;

    &:hover {
      box-shadow: 0px -1px 5px 2px rgba(255,255,255,1);
    }
  `;

  if (history.location.pathname === '/') return (<span />);

  return (
    <ContainerStyled onClick={(): void => history.goBack()}>
      Go back
      <i 
        className="material-icons"
        css={css`
          color: #FFF;
          font-size: 36px;
        `}
      >
        keyboard_backspace
      </i>
    </ContainerStyled>
  );
};

export default GoBackButtonComponent;
