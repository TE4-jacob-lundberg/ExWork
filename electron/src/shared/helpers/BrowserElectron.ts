// eslint-disable-next-line @typescript-eslint/no-explicit-any
export let electron: any;

if (process.env.ENVIRONMENT === 'test') electron = require('electron');
else electron = window.require('electron');
