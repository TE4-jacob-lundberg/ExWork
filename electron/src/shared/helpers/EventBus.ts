/* eslint-disable @typescript-eslint/no-explicit-any */
interface IEventBus {
  events: {[id: string]: Function[]};
  dispatch: (event: string, data: any) => void;
  subscribe: (event: string, callBack: Function) => void;
}

const EventBus: IEventBus = {
  events: {},
  dispatch: function (event: string, data: any): void {
    if (!this.events[event]) return;
    this.events[event].forEach((callBack: Function) => callBack(data));
  },
  subscribe: function (event: string, callBack: Function): void {
    if (!this.events[event]) this.events[event] = [];
    this.events[event].push(callBack);
  },
};

export { EventBus };
