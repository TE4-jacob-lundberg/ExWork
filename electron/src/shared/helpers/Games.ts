import { standardGames, IGame } from '../constants/standardGames';

export class Games {

  public static all(): IGame[] {
    if (!localStorage.getItem('games')) localStorage.setItem('games', JSON.stringify(standardGames));

    return JSON.parse(localStorage.getItem('games')!) || [];
  }

  public static addNew(game: IGame): string {
    const savedGames: IGame[] = JSON.parse(localStorage.getItem('games')!);
    if (savedGames.some(sGame => sGame.id === game.id)) return 'DUPLICATE_ID';
    
    localStorage.setItem('games', JSON.stringify(savedGames.concat(game)));
    return '';
  }
}
