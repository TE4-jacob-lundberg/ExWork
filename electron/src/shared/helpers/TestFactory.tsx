/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';

export function factory(component: JSX.Element): RenderResult {
  return render(component);
}

export function routerFactory(component: JSX.Element, options: any = { history: createBrowserHistory() }): RenderResult {
  return factory(
    <Router history={options.history}>
      {component}
    </Router>
  );
}
