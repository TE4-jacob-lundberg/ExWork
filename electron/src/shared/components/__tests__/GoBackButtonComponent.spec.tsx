import React from 'react';
import { fireEvent } from '@testing-library/react';
import { createMemoryHistory } from 'history';

import { routerFactory } from '../../helpers/TestFactory';
import GoBackButtonComponent from '../GoBackButtonComponent';

const testComponent = <GoBackButtonComponent />;

describe('GoBackButtonComponent.tsx', () => {
  it('renders correctly', () => {
    const history = createMemoryHistory({initialEntries: ['/test']});
    const { container } = routerFactory(testComponent, {history});
    
    expect(container).toMatchSnapshot();
  });

  it('goes back in history on click', () => {
    const history = createMemoryHistory({initialEntries: ['/']});
    history.push('/test');
    const { getByText } = routerFactory(testComponent, {history});

    expect(history.location.pathname).toBe('/test');

    fireEvent.click(getByText('Go back'));

    expect(history.location.pathname).toBe('/');
  });

  it('doesn\'t show on root route', () => {
    const history = createMemoryHistory({initialEntries: ['/test']});
    const { container } = routerFactory(testComponent, {history});

    expect(container).toMatchSnapshot();

    history.push('/');

    expect(container).toMatchSnapshot();
  });
});
