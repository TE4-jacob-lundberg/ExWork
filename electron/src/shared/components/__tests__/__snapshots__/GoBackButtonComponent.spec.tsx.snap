// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoBackButtonComponent.tsx doesn't show on root route 1`] = `
<div>
  <div
    class="css-1lkolb"
  >
    Go back
    <i
      class="material-icons css-oqb07f-GoBackButtonComponent"
    >
      keyboard_backspace
    </i>
  </div>
</div>
`;

exports[`GoBackButtonComponent.tsx doesn't show on root route 2`] = `
<div>
  <span />
</div>
`;

exports[`GoBackButtonComponent.tsx renders correctly 1`] = `
<div>
  <div
    class="css-1lkolb"
  >
    Go back
    <i
      class="material-icons css-oqb07f-GoBackButtonComponent"
    >
      keyboard_backspace
    </i>
  </div>
</div>
`;
